runner/internal/shim/api/http.go

```diff
@@ -51,3 +51,20 @@ func (s *ShimServer) PullGetHandler(w http.ResponseWriter, r *http.Request) (int
 		State: string(s.runner.GetState()),
 	}, nil
 }
+
+func (s *ShimServer) StopPostHandler(w http.ResponseWriter, r *http.Request) (interface{}, error) {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+
+	var body SubmitStopBody
+	if err := api.DecodeJSONBody(w, r, &body, true); err != nil {
+		log.Println("Failed to decode submit stop body", "err", err)
+		return nil, err
+	}
+
+	s.runner.Stop(body.Force)
```

Comment:
Why don't we show errors to the user, if any?

Reply:
I'll do it in next iteration.